networking_cisco/plugins/cisco/cfg_agent/service_helpers/routing_svc_helper.py

```diff
@@ -670,19 +722,12 @@ def _process_router(self, ri):
             if ri.router[ROUTER_ROLE_ATTR] not in \
                     [c_constants.ROUTER_ROLE_GLOBAL,
                      c_constants.ROUTER_ROLE_LOGICAL_GLOBAL]:
-                if not ri.router['admin_state_up']:
-                    self._disable_router_interface(ri)
-                else:
-                    if ex_gw_port:
-                        if not ex_gw_port['admin_state_up']:
-                            self._disable_router_interface(ri, ex_gw_port)
-                        else:
-                            self._enable_router_interface(ri, ex_gw_port)
-                    for port in internal_ports:
-                        if not port['admin_state_up']:
-                            self._disable_router_interface(ri, port)
-                        else:
-                            self._enable_router_interface(ri, port)
+                internal_ports = ri.router.get(l3_constants.INTERFACE_KEY, [])
```

Comment:
this u have queried before inside _get_internal_port_changes()....so can u just return internal_ports in _get_internal_port_changes() also to avoid the same query twice?